person.rb

```diff
@@ -19,10 +26,39 @@ def can_use_services?
     end
   end
 
+  def correct_name
+    @name
+  end
+
   private
 
   def is_of_age
     @age >= 18
   end
   # rubocop:enable Naming/PredicateName
 end
+
+class BaseDecorator < Namable
+  attr_accessor :namable
+
+  def initialize(namable)
+    super()
+    @namable = namable
+  end
+
+  def correct_name
+    @namable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @namable.correct_name.upcase
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @namable.correct_name.strip if @namable.correct_name.length <= 10
+  end
```

Comment:
- [ ] This method is meant to check that the name has a maximum of 10 characters. If it's longer it should trim the word. However, the strip method you are using returns a copy of the original name with trailing and leading whitespace removed but it doesn't check that the name has a maximum of 10 characters. Kindly fix this.

```suggestion
  def correct_name
    if @namable.correct_name.length > 10
       @nameable.correct_name[0..9]
    else
        @nameable.correct_name
  end
```